props/saber_sabersense_buttons.h

```diff
@@ -275,27 +288,53 @@ GESTURE CONTROLS
 #define PROPS_SABER_SABERSENSE_BUTTONS_H
 
 #ifdef SABERSENSE_ARRAY_SELECTOR
+
+#ifndef SABERSENSE_DEFAULT_ARRAY
+#define SABERSENSE_DEFAULT_ARRAY 1
+#endif
+
+  // Check that user-specified SABERSENSE_DEFAULT_ARRAY is within valid range.
+  constexpr size_t NUM_ARRAYS = sizeof(blades) / sizeof(blades[0]);
+  static_assert(SABERSENSE_DEFAULT_ARRAY >= 1 && SABERSENSE_DEFAULT_ARRAY <= NUM_ARRAYS,
+
+    // Show error below if array number is invalid. (Returns and indents for IDE readability).
+    "\n  ERROR:\n    SABERSENSE_DEFAULT_ARRAY number invalid. Ensure array number specified "
+    "does not exceed number of blade arrays in config.\n "
+    "   Arrays should be numbered sequentially. First array should be zero (or NO_BLADE if "
+    "using Blade Detect) then 1, 2, 3, etc.\n");
+
 #ifndef SABERSENSE_DISABLE_SAVE_ARRAY
 class SaveArrayStateFile : public ConfigFile {
 public:
   void iterateVariables(VariableOP *op) override {
-    CONFIG_VARIABLE2(sabersense_array_index, 0);  // Default array if no save file.
+
+#ifndef BLADE_DETECT_PIN
+    // Minus 1 allows for extra NO_BLADE array for consistent numbering for end user.
+    CONFIG_VARIABLE2(sabersense_array_index, (SABERSENSE_DEFAULT_ARRAY - 1));
+#else   
+    CONFIG_VARIABLE2(sabersense_array_index, SABERSENSE_DEFAULT_ARRAY);
+#endif
   }
-    int sabersense_array_index;  // Stores current array index.
+    int sabersense_array_index;
   };
 #endif
 
     struct SabersenseArraySelector {
-      static int return_value;  // Tracks current array index.
+      static int return_value;
       float id() {
         return return_value;
       }
-      static void cycle() {
-        return_value = (return_value + 1) % NELEM(blades);
-      }
-    };
-    int SabersenseArraySelector::return_value;
-    
+
+    static void cycle() {
+      int i = return_value;
```

Comment:
if return_value is an ohm value, you probably want to scan the array to find the index of the entry that has the given return value, like maybe:

```
int i = 0; for(;i < (int)NELEM(blades); i++) if (blades[i].ohm == return_value) break;
```